lisa/base_tools/wget.py

```diff
@@ -64,18 +72,26 @@ def get(
             command = f"{command} -O {download_path}"
         else:
             command = f"{command} -P {download_path}"
-        command_result = self.run(
-            command,
-            no_error_log=True,
-            shell=True,
-            sudo=sudo,
-            force_run=force_run,
-            timeout=timeout,
-        )
-        matched_result = self.__pattern_path.match(command_result.stdout)
-        if matched_result:
-            download_file_path = matched_result.group("path")
-        else:
+
+        retry = 5
+        download_file_path = ""
+        while retry:
+            command_result = self.run(
+                command,
+                no_error_log=True,
+                shell=True,
+                sudo=sudo,
+                force_run=force_run,
+                timeout=timeout,
+            )
+            download_file_path = self._get_downladed_file_path(command_result.stdout)
+            if download_file_path:
+                break
+            else:
+                retry -= 1
+                time.sleep(5)
+
+        if not download_file_path:
```

Comment:
You can add a retry decorator on this method.

Reply:
That was giving me a type checking warning, I just noticed it's used elsewhere so maybe it's alright? I'll switch.